spec.html

```diff
@@ -39502,15 +39501,11 @@ <h1>
             1. Let _srcByteOffset_ be _srcArray_.[[ByteOffset]].
             1. Let _elementLength_ be _srcArray_.[[ArrayLength]].
             1. Let _byteLength_ be _elementSize_ &times; _elementLength_.
-            1. If IsSharedArrayBuffer(_srcData_) is *false*, then
-              1. Let _bufferConstructor_ be ? SpeciesConstructor(_srcData_, %ArrayBuffer%).
-            1. Else,
-              1. Let _bufferConstructor_ be %ArrayBuffer%.
             1. If _elementType_ is the same as _srcType_, then
-              1. Let _data_ be ? CloneArrayBuffer(_srcData_, _srcByteOffset_, _byteLength_, _bufferConstructor_).
+              1. Let _data_ be ? CloneArrayBuffer(_srcData_, _srcByteOffset_, _byteLength_).
             1. Else,
-              1. Let _data_ be ? AllocateArrayBuffer(_bufferConstructor_, _byteLength_).
-              1. If IsDetachedBuffer(_srcData_) is *true*, throw a *TypeError* exception.
+              1. Let _data_ be ? <emu-meta suppress-effects="user-code">AllocateArrayBuffer(%ArrayBuffer%, _byteLength_)</emu-meta>.
+              1. Assert: IsDetachedBuffer(_srcData_) is *false*.
```

Comment:
I don't think this assertion really adds anything; there's a detachedness check above and no user code runs between then and here. Also the only place it's going to be consumed is as an argument to `GetValueFromBuffer`, which has this as an assertion itself.

Reply:
Removed.